### ISSUE TITLE
Miniseed file not fully read

### ISSUE THREAD

**Issue by raciner** [obspy/obspy#3447]
### Avoid duplicates

- [X] I searched existing issues

### Bug Summary

The file linked here: [https://polybox.ethz.ch/index.php/s/vjpoaOnbzQ8hfaH] contains data from 2024-05-14T23:52:07 to 2024-05-16T00:04:33. However, when using 

st = obspy.core.read('file.mseed')
print(st)

this is shown:
XX.XDR01.02.VKD | 2024-05-14T23:52:07.000000Z - 2024-05-15T23:57:37.000000Z | 0.1 Hz, 8674 samples

The problem is that the delta is slightly higher than 10.0 due to the behaviour of the datalogger, however, the difference is within the default tolerance of obspy, so that the difference between the time used in obspy and the actual timestamp grows up to ten minutes throughout the day. The time record for each individual miniseed record is correct. Other software such as qmerge handles this differently. Wouldn't it make more sense to accumulate the time difference between the time indicated in the header and the time that obspy computes based on the time stamp of the first header + delta*number of seen data points and introduce a break once this gets bigger than the tolerance instead of not accumulating it and checking it for each boundary between records individually?

### ObsPy Version?

1.4.1

### Python Version?

3.10.12

### Installation Method?

pip

**Comment by megies** [comment]
I'm a little bit confused. From what I can see the sampling interval is stated in the file as exactly 10s but the individual records' start times don't line up, i.e. there are gaps and overlaps on a subsample scale between every record? That correct or am I missing something?

We are merging together records that have gap/overlaps on subsample scale, otherwise you would end up with 420 individual `Trace` objects.

Am I correct that you want obspy to assume the file is without gaps and compute an odd sampling rate and treat it as a gapless trace?

I'm pretty sure that would mean touching our [C code wrapper for libmseed around here](https://github.com/obspy/obspy/blob/84d4ca41d902609ab758661bc23ad3e8d4ec3ea4/obspy/io/mseed/src/obspy-readbuffer.c#L560-L569), maybe passing in a flag from Python to change behavior in the wrapper. I'm no C wizard, so definitely somebody else would need to do that.

**Comment by megies** [comment]
> accumulate the time difference between the time indicated in the header and the time that obspy computes based on the time stamp of the first header + delta*number of seen data points and introduce a break once this gets bigger than the tolerance instead of not accumulating it and checking it for each boundary between records individually?

could make sense as a switch, but this would mean adding a lot of logic to our libmseed wrapper code and somebody else would have to do it.

**Comment by raciner** [comment]
The basic issue is that the datalogger is supposed to make one measurement every 10s. However, due to the poor quality its implementation, this is not stable, so that there is a slow drift. From one miniseed record to the next, it's always within the tolerance, but on a whole day, it adds up to about seven minutes, which means, that the timestamps are completely wrong after reading the file and writing it to disk. A less surprising behaviour would be to declare a gap or overlap whenever the accumulated time difference is larger than the tolerance.

**Comment by megies** [comment]
> A less surprising behaviour would be to declare a gap or overlap whenever the accumulated time difference is larger than the tolerance

I agree that makes sense, I might not be able to work on it anytime soon though, trying to finish up some loose ends and more of those appear than get closed anyway..
